### PR TITLE
Added a pytorch implementation of OmniCV's to remove equilib package usage

### PR DESCRIPTION
**Pull request by THE-COB**
The equilib package was under a GPL so we couldn't use it in nerfstudio since nerfstudio is MIT. I just implemented the equirectangular to perspective method from [this MIT licensed library](https://github.com/kaustubh-sadekar/OmniCV-Lib) in pytorch and then used it instead.

### CODE REVIEW

File: nerfstudio/process_data/equirect_utils.py [hunk 1]
```diff
@@ -22,18 +22,139 @@
...
-    TextColumn,
-    TimeRemainingColumn,
-)
+from rich.progress import BarColumn, Progress, TaskProgressColumn, TextColumn, TimeRemainingColumn
 
 from nerfstudio.utils.rich_utils import CONSOLE, ItersPerSecColumn
 
 
+# https://gist.github.com/fgolemo/94b5caf0e209a6e71ab0ce2d75ad3ed8
+def rotation_matrix(axis: torch.Tensor, theta: torch.float) -> torch.Tensor:
```
Comment (tancik) [inline nerfstudio/process_data/equirect_utils.py:31]: I think this functionality already exists - https://github.com/nerfstudio-project/nerfstudio/blob/53c7c8eb07700ab087d36d18665175b39ab093c2/nerfstudio/cameras/camera_utils.py#L451

Comment (THE-COB) [reply nerfstudio/process_data/equirect_utils.py:31]: oh awesome, I'll use that

File: nerfstudio/process_data/equirect_utils.py [hunk 2]
```diff
@@ -22,18 +22,139 @@
...
+    """
+    Generalized 3d rotation via Euler-Rodriguez formula, https://www.wikiwand.com/en/Euler%E2%80%93Rodrigues_formula
+    Return the rotation matrix associated with counterclockwise rotation about
+    the given axis by theta radians.
+
+    axis: torch.Tensor, shape (3,)
+    theta: torch.float
+
+    return: torch.Tensor, shape (3, 3)
```
[the remark this answers is not in the document]
Comment (THE-COB) [reply nerfstudio/process_data/equirect_utils.py:40]: oh oops I'll fix that soon

File: nerfstudio/process_data/equirect_utils.py [hunk 3]
```diff
@@ -22,18 +22,139 @@
...
+    pytorch reimlement of https://github.com/kaustubh-sadekar/OmniCV-Lib
+
+    img: torch.Tensor, shape (batch_size, channels, height, width)
+    fov: int, field of view
+    theta: int, left/right angle
+    phi: int, up/down angle
+    hd: int, height of perspective image
+    wd: int, width of perspective image
+
+    return: torch.Tensor, shape (batch_size, channels, hd, wd)
```
[the remark this answers is not in the document]
Comment (THE-COB) [reply nerfstudio/process_data/equirect_utils.py:96]: will fix

File: nerfstudio/process_data/equirect_utils.py (hunk 1 above)
Comment (THE-COB) [reply nerfstudio/process_data/equirect_utils.py]: so it turns out the functions were different enough so I just kept the new one but renamed it to decrease confusion. The new one implements Euler-Rodriquez rotation.

**Review by tancik — APPROVED** [review]
Lgtm

**Comment by tancik** [comment]
Looks good, but need to remove the library from the dependency list.

**Comment by THE-COB** [comment]
> Looks good, but need to remove the library from the dependency list.

done

**Review by tancik — APPROVED** [review]

**Pull request merged**